Documented the use of the "subdirectory" parameter (#5949)
python-poetry/poetry: docs/dependency-specification.md

numpy = { git = "https://github.com/numpy/numpy.git", tag = "v0.13.2" }
```

> In cases where the package you want to install is located in a subdirectory of the VCS repo, you can use the subdirectory option:

Reviewer: ```suggestion
In cases where the package you want to install is located in a subdirectory of the VCS repository, you can use the `subdirectory` option, similarly to what [pip](https://pip.pypa.io/en/stable/topics/vcs-support/#url-fragments) provides:
```

```toml
[tool.poetry.dependencies]

> # Install a package named `subdir_package` from a folder called `subdir` within the repo

Reviewer: ```suggestion
# Install a package named `subdir_package` from a folder called `subdir` within the repository
```

> subdir_package = { git = "https://github.com/mypackage_with_subdirs.git", subdirectory = "subdir" }

Reviewer: ```suggestion
subdir_package = { git = "https://github.com/myorg/mypackage_with_subdirs.git", subdirectory = "subdir" }
```

```

> When using `poetry add` to achive the same options, you can use the same URL formatting that is also supported by [pip](https://pip.pypa.io/en/stable/topics/vcs-support/#url-fragments).

Reviewer: ```suggestion
with the corresponding `add` call:
```
to be consistent with https://github.com/python-poetry/poetry/blob/a3aafa840de950f81e99553e52190a54c94d6bce/docs/dependency-specification.md?plain=1#L213 and https://github.com/python-poetry/poetry/blob/a3aafa840de950f81e99553e52190a54c94d6bce/docs/dependency-specification.md?plain=1#L249

```bash

> poetry add "https://github.com/mypackage_with_subdirs.git#subdirectory=subdir"

Reviewer: ```suggestion
poetry add "https://github.com/myorg/mypackage_with_subdirs.git#subdirectory=subdir"
```

poetry add git+https://github.com/org/project.git
```

> In cases where the package you want to install is located in a subdirectory of the VCS repo, you can use the subdirectory option:

Reviewer 2: This whole added section should be moved higher (line 147 maybe?)  since now it interferes with information about git configuration.

Author: Agreed! Changes it accordingly. Thanks for the review
